### PR TITLE
Apply radius scale in the editor

### PR DESCRIPTION
**Pull request by jameskoster**
## What?
This PR applies the radius scale to the block editor UI.

> [!NOTE]  
> As it is something of a branding element I've intentionally left the block toolbar out of this work. It warrants isolated consideration.

The changes typically fall into one of the following categories...

### Adopting the scale / visual changes

Where relevant, elements have been modified to adopt the [radius scale](https://github.com/WordPress/gutenberg/issues/63703). In terms of visual changes there weren't too many cases in this category because the components package already adopted the new scale in https://github.com/WordPress/gutenberg/pull/64368. One example is anchors in list view:

Screenshot 2024-08-30 at 13 08 15

Otherwise the vast majority of updates here involve a straight swap from `$radius-block-ui` to `radius-small`.

### Removing obsolete style declarations

Working through the code I found a few instances of `border-radius` being applied to elements which did not visually appear in the UI, most likely because the design has changed since the original implementation. Instead of leaving the obsolete code in place I removed it.

Similarly, there were numerous instances where `border-radius` was unnecessarily duplicated as a style override, even though the underlying component already provided the required styling. Buttons and Inputs were the main offenders here. Again, instead of leaving the obsolete code I removed it.

### Miscellaneous visual changes
In https://github.com/WordPress/gutenberg/pull/60757 block outlines were updated to remove rounded corners. However, a number of on-canvas outlines remained rounded which felt inconsistent. One example is the dashed outline one finds in empty Row/Stack blocks. In the interest of consistency I removed these radii.

Screenshot 2024-08-30 at 13 23 51

## Testing Instructions
1. Open the site editor
2. Browse around all sections, including the full screen editor
3. Look out for any changes that do not fit the radius scale

### CODE REVIEW

File: packages/dataviews/src/dataviews-layouts/list/style.scss [hunk 1]
```diff
@@ -114,12 +114,12 @@ ul.dataviews-view-list {
 			&::before {
 				position: absolute;
 				content: "";
-				top: calc(var(--wp-admin-border-width-focus) + 1px);
+				top: var(--wp-admin-border-width-focus);
```
Comment (tyxla) [inline packages/dataviews/src/dataviews-layouts/list/style.scss:117]: 🤔 Was this change intended for another PR?

File: packages/edit-site/src/components/layout/style.scss [hunk 2]
```diff
@@ -200,7 +200,7 @@ html.canvas-mode-edit-transition::view-transition-group(toggle) {
 		right: 9px;
 		bottom: 9px;
 		left: 9px;
-		border-radius: $radius-block-ui + $border-width + $border-width;
+		border-radius: $radius-medium;
```
Comment (tyxla) [inline packages/edit-site/src/components/layout/style.scss:203]: Wow, that was akward; thanks for fixing!

File: packages/edit-site/src/components/layout/style.scss [hunk 3]
```diff
@@ -120,7 +120,7 @@
 			overflow: hidden;
 
 			.edit-site-layout:not(.is-full-canvas) & {
-				border-radius: $radius-block-ui * 4;
+				border-radius: $radius-large;
```
Comment (tyxla) [inline packages/edit-site/src/components/layout/style.scss:123]: 👍

File: packages/editor/src/components/resizable-editor/style.scss [hunk 4]
```diff
@@ -16,7 +16,7 @@
 	cursor: ew-resize;
 	outline: none;
 	background: none;
-	border-radius: $radius-block-ui;
+	border-radius: $radius-full;
```
Comment (tyxla) [inline packages/editor/src/components/resizable-editor/style.scss:19]: I'm assuming we're intentionally going with `radius-full` here?

File: packages/editor/src/components/post-locked-modal/style.scss [hunk 5]
```diff
@@ -3,7 +3,7 @@
 }
 
 .editor-post-locked-modal__avatar {
-	border-radius: $radius-block-ui;
+	border-radius: $radius-round;
```
Comment (tyxla) [inline packages/editor/src/components/post-locked-modal/style.scss:6]: Round makes sense here 👍

**Review by tyxla — APPROVED** [review]
Tested thoroughly, everything looks good 👍 

Code looks alright too, I've seen some changes but all made sense in terms of removing unnecessary styles or changing styles for consistency.

Left a few questions, but from my perspective this is good to go 👍

File: packages/dataviews/src/dataviews-layouts/list/style.scss (hunk 1 above)
Comment (jameskoster) [reply packages/dataviews/src/dataviews-layouts/list/style.scss:117]: This one is tiny visual bug I observed while testing. Figured it was okay to include here, but can also separate it out if y'all prefer.

File: packages/editor/src/components/resizable-editor/style.scss (hunk 4 above)
Comment (jameskoster) [reply packages/editor/src/components/resizable-editor/style.scss:19]: Yes, this is consistent with similar elements in the components package that we want to be fully rounded. E.g: https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/progress-bar/styles.ts#L43

File: packages/dataviews/src/dataviews-layouts/list/style.scss (hunk 1 above)
Comment (tyxla) [reply packages/dataviews/src/dataviews-layouts/list/style.scss:117]: Up to you, really. I'd prefer a separate PR, but no strong feelings either way, so leaving to your judgment.

**Comment by jasmussen** [comment]
I'm happy to land this as an improvement. The corners of segmented control active items get a bit sharp though, at 1px, and they diverge from the 2px that are applied to toggle buttons (i.e. bold, etc). I recognize there's a separate conversation about the scale at play there, though.

**Comment by jameskoster** [comment]
The toggle button active state has a 1px radius for neater visual nesting. We can increase/decrease it of course, but that will mean tweaking other values in the scale. Once this PR is merged we can open a new one to try that very quickly; it's just two files to edit now rather than hundreds as before.

> they diverge from the 2px that are applied to toggle buttons (i.e. bold, etc)

The block toolbar needs updating, but as noted in the OP that particular UI should be tackled independently in a separate PR.

**Comment by jameskoster** [comment]
I'm going to merge this one. There's an enormous amount of tidy up included, and the earlier we merge before beta the more likely we are to spot any outlying problems.
